src/Parentheses/Solution.java

```diff
@@ -0,0 +1,74 @@
+package Parentheses;
+
+import java.util.Stack;
+
+public class Solution {
+
+    public boolean isValid(String parentheses) {
+
+        char[] arrayParentheses = parentheses.toCharArray();            // O(parentheses)
+        int allOk = 0;                                                  // O(1)
+                                                                       // time          Space
+        Stack<Character> pila = new Stack<>(); // O(1)
+        for (int i = 0; i < arrayParentheses.length; i++) {            // O(arrayParentheseslength);  O(1)
+            char character = arrayParentheses[i];                      // O(1);         O(1)
+
+            if (character == '[' ||                                     // O(1)         O(0)
+                    character == '{' ||
+                    character == '(') {
+                pila.push(character);                                   // O(1)         O(1)
+                System.out.println("pila: " + pila);
+            } else {
+                closeParentheses(pila);
+
+                switch (character) {                                        // O(1)
+                    case ')':                                               // O(1)
+                        if (pila.isEmpty()) {                               // O(1)     O(0)
+                            allOk = 1;                                      // O(1)
+                        } else {
+                            char openCharacter = '(';                       // O(1)
+                            if (pila.pop() != openCharacter) {              // O(1)     O(1)
+                                allOk = 1;                                  // O(1)
+                            }
+                            System.out.println("pila: " + i + " " + pila);
+                        }
+                        break;
+
+                    case '}':
+                        if (pila.isEmpty()) {                                   // O(1)         O(0)
+                            allOk = 1;
+                        } else {
+                            char openCharacter = '{';
+                            if (pila.pop() != openCharacter) {                  // O(1)         O(1)
+                                allOk = 1;
+                            }
+                            System.out.println("pila: " + i + " " + pila);
+                        }
+                        break;
+
+                    case ']':
+                        if (pila.isEmpty()) {                                   // O(1)     O(0)
+                            allOk = 1;
+                        } else {
+                            char openCharacter = '[';
+                            if (pila.pop() != openCharacter) {                  // O(1)     O(1)
+                                allOk = 1;
+                            }
+                            System.out.println("pila: " + i + " " + pila);
+                        }
```

Comment:
It looks like the body of each case could be extracted into a method. Then we could just call that method with different arguments. For example:
```java
checkNextParenthesis(pila, openCharacter)
```